core/src/main/scala/zio/logging/Logging.scala

```diff
@@ -0,0 +1,39 @@
+package zio.logging
+
+import zio.{ Cause, FiberRef, UIO, ZIO }
+
+trait AbstractLogging[Message] {
+  def logging: AbstractLogging.Service[Any, Message]
+}
+
+object AbstractLogging {
+
+  trait Service[-R, Message] {
+    def context: LoggingContext[Message]
+    def trace(message: Message): ZIO[R, Nothing, Unit]
```

Comment:
Should `message` be lazy? So that it does not need to be generated when the logger is not enabled for `trace`?

Reply:
good point

Reply:
done